src/headers.js

```diff
@@ -35,68 +44,103 @@ function fill (headers, object) {
 
 class Headers {
   constructor (init) {
-    this[kHeaders] = new Map()
+    this[kHeaders] = []
 
     if (init && typeof init === 'object') {
       fill(this, init)
     }
   }
 
   append (name, value) {
-    const [normalizedHeaderName, normalizedHeaderValue] = normalizeAndValidateHeaderValue(name, value)
-
-    const existing = this[kHeaders].get(normalizedHeaderName)
-    const newHeaderValue = existing ? existing + ', ' + normalizedHeaderValue : normalizedHeaderValue
-    this[kHeaders].set(normalizedHeaderName, newHeaderValue)
+    const [normalizedName, normalizedValue] = normalizeAndValidateHeaderValue(name, value)
+
+    let index = this[kHeaders].length
+    for (let i = 0; i < this[kHeaders].length; i += 2) {
+      if (normalizedName === this[kHeaders][i]) {
+        this[kHeaders][i + 1] += `, ${normalizedValue}`
+        return
+      } else if (this[kHeaders][i] > normalizedName) {
+        index = i
+        break
+      }
+    }
```

Comment:
Have you tried with a binary search?

Reply:
have not, i'll give that a shot next

Reply:
Binary search also crossed my mind -- https://www.tbray.org/ongoing/When/200x/2003/03/22/Binary

Reply:
How does sort on insert compare to the previous iteration? @Ethan-Arrowood

Reply:
sort on insert is faster. taking a look at your binary sort now :)